packages/cli/src/lib/version-check/update-process.js

```diff
@@ -0,0 +1,53 @@
+import { exec } from 'child_process'
+
+import {
+  resetHistory,
+  setVersions,
+  initStorage,
+  getVersions,
+} from './storage.js'
+
+function indexOfColumn(table, name) {
+  return table.data.head.indexOf(name)
+}
+
+function tableToVersions(table) {
+  const releavntColumns = ['Current', 'Latest']
+  const packageName = '@redwoodjs/core'
+  const packageColumnName = 'Package'
+  const redwoodPackageInfo = table.data.body.find(function (packageArr) {
+    return packageName === packageArr[indexOfColumn(table, packageColumnName)]
+  })
+  return releavntColumns.reduce(function (acc, tableKeyName) {
+    acc[tableKeyName.toLowerCase()] =
+      redwoodPackageInfo[indexOfColumn(table, tableKeyName)]
+    return acc
+  }, {})
+}
+
+function parseYarnOutdatedResponse(stdout) {
+  try {
+    // The return string is actually two lines of JSON objects, with the second being the table with the relevant info
+    const table = JSON.parse(stdout.split('\n')[1])
+    return tableToVersions(table)
+  } catch (e) {
+    console.log(e)
+    return undefined
+  }
+}
+
+exec('yarn outdated --json', async (err, stdout) => {
+  // yarn outdated returning an error means there is at least one outdated package
+  if (typeof err === 'object' && typeof stdout === 'string') {
+    const versions = parseYarnOutdatedResponse(stdout)
+    if (versions) {
+      await initStorage()
+      const prevVerious = await getVersions()
+      // only update when there is a newer version
+      if (!prevVerious || prevVerious.latest !== versions.latest) {
+        await resetHistory()
+        await setVersions(versions)
+      }
+    }
+  }
+})
```

Comment:
I haven't tested this locally yet -- curious if you've tried it yet with `canary` or `rc` packages and whether or not this has unintended effect on behavior.

Background (just in case): we automatically publish `canary` versions with each merged PR which you can see on [NPMJS on versions tab](https://www.npmjs.com/package/@redwoodjs/core). And users can upgrade easily via `yarn rw upgrade --tag canary`

Reply:
I hadn't considered that and will test it this week. It seems like the prompt should only show when the currently installed version is less than the latest published version.